apps/src/studio/studio.js

```diff
@@ -2748,7 +2748,8 @@ Studio.runButtonClick = function () {
 Studio.displayFeedback = function () {
   var tryAgainText;
   // For free play, show keep playing, unless it's a big game level
-  if (level.freePlay && !(Studio.customLogic instanceof BigGameLogic)) {
+  if (Studio.testResults >= TestResults.MINIMUM_OPTIMAL_RESULT &&
+      !(Studio.customLogic instanceof BigGameLogic)) {
```

Comment:
nit: I think I'd still like to see `level.freePlay` as an explicit check, even though the testresults comparison will still work for that case. Just for clarity.